src/clients/c++/perf_client.cc

```diff
@@ -181,6 +181,25 @@ typedef struct PerformanceStatusStruct {
 
 enum ProtocolType { HTTP = 0, GRPC = 1 };
 
+nic::Error
+ReadFile(const std::string& path, std::vector<char>* contents)
+{
+  std::ifstream in(path, std::ios::in | std::ios::binary);
+  if (!in) {
+    return nic::Error(
+        ni::RequestStatusCode::INVALID_ARG,
+        "failed to open file '" + path + "'");
+  }
+
+  in.seekg(0, std::ios::end);
+  contents->resize(in.tellg());
+  in.seekg(0, std::ios::beg);
+  in.read(&(*contents)[0], contents->size());
```

Comment:
If `path` points to an empty file, the vector size will be zero.  Is it valid to take the address of a non-existent 0th element?

(It's been many years since I worked in C++)

Reply:
Added check on return value of `in.tellg()`